.gitmodules

```diff
@@ -4,3 +4,6 @@
 [submodule "mini-gdbstub"]
 	path = src/mini-gdbstub
 	url = https://github.com/RinHizakura/mini-gdbstub
+[submodule "softfloat"]
+	path = src/softfloat
+	url = https://github.com/visitorckw/SoftFloat-3e.git
```

Comment:
Use shorter URL. i.e., https://github.com/visitorckw/SoftFloat-3e`
Can you utilize https://github.com/ucb-bar/berkeley-softfloat-3 instead?

Reply:
The Makefile for building softfloat.a at https://github.com/ucb-bar/berkeley-softfloat-3 doesn't specify an output path parameter. As a result, using it may require keeping softfloat.a within the src/softfloat instead of the build/softfloat. Additionally, the previously mentioned issue of lengthy header inclusion in riscv.h might be temporarily challenging to resolve.

Reply:
Rather than duplicating files without their historical commits, consider forking `ucb-bar/berkeley-softfloat-3` and incorporating your own modifications and bug fixes. Does this approach seem reasonable?

Reply:
Agreed. I'll make the adjustment.